pkg/api/v20191231preview/openshiftcluster_validatestatic_test.go

```diff
@@ -699,6 +700,11 @@ func TestOpenShiftClusterStaticValidateDelta(t *testing.T) {
 			},
 			wantErr: "400: PropertyChangeNotAllowed: properties.masterProfile.subnetId: Changing property 'properties.masterProfile.subnetId' is not allowed.",
 		},
+		{
+			name:    "worker name change",
+			modify:  func(oc *OpenShiftCluster) { oc.Properties.WorkerProfiles[0].Name = "new-name" },
+			wantErr: "400: PropertyChangeNotAllowed: properties.workerProfiles['new-name'].name: Changing property 'properties.workerProfiles['new-name'].name' is not allowed.",
```

Comment:
Why does `immutable.Validate()` produces errors where istead of index it prints a value from the `Name` field?

https://github.com/Azure/ARO-RP/blob/389628e3b697d34a449e2b795258c5cdca43a169/pkg/util/immutable/immutable.go#L86-L91

Reply:
as a sop to usability.  If the Name field doesn't exist, it reverts to index.